thirdeye/thirdeye-frontend/app/utils/manage-alert-utils.js

```diff
@@ -175,46 +176,67 @@ export function evalObj() {
+ * @param {Object} graphConfig - the metric settings
+ * @returns {String} metric data call params/url
+ */
+export function buildMetricDataUrl(graphConfig) {
+  const { id, maxTime, filters, dimension, granularity } = graphConfig;
+  const selectedDimension = dimension ? dimension : 'All';
+  const startTimeBucket = granularity && granularity.toLowerCase().includes('minute') ? 'week' : 'months';
+  const currentEnd = moment(maxTime).isValid() ? moment(maxTime).valueOf() : buildDateEod(1, 'day').valueOf();
+  const currentStart = moment(currentEnd).subtract(1, startTimeBucket).valueOf();
+  const baselineStart = moment(currentStart).subtract(1, 'week').valueOf();
+  const baselineEnd = moment(currentEnd).subtract(1, 'week');
+
+  return `/timeseries/compare/${id}/${currentStart}/${currentEnd}/${baselineStart}/${baselineEnd}?dimension=` +
+         `${selectedDimension}&granularity=${granularity}&filters=${encodeURIComponent(filters)}`;
+}
+
 /**
  * If a dimension has been selected, the metric data object will contain subdimensions.
- * This method calls for dimension ranking by metric, filters for the selected dimension,
- * and returns a sorted list of graph-ready dimension objects.
+ * This method averages each subdimension's total change rate and returns a sorted list
+ * of the top X graph-ready dimension objects
  * @method getTopDimensions
- * @param {Object} dimensionObj - the object containing available subdimension for current metric
- * @param {Array} scoredDimensions - array of dimensions scored by relevance
- * @param {String} selectedDimension - the user-selected dimension to graph
- * @return {RSVP Promise}
+ * @param {Object} metricData - the graphable metric data returned from fetchAnomalyGraphData()
+ * @param {Number} dimCount - number of dimensions to allow in response
+ * @return {undefined}
  */
-export function getTopDimensions(dimensionObj = {}, scoredDimensions, selectedDimension) {
-  const maxSize = 5;
+export function getTopDimensions(metricData, dimCount) {
   const colors = ['orange', 'teal', 'purple', 'red', 'green', 'pink'];
+  const dimensionObj = metricData.subDimensionContributionMap || {};
+  const dimensionKeys = Object.keys(dimensionObj);
+  let processedDimensions = [];
   let dimensionList = [];
   let colorIndex = 0;
 
-  if (selectedDimension) {
-    const filteredDimensions =  _.filter(scoredDimensions, (dimension) => {
-      return dimension.label.split('=')[0] === selectedDimension;
-    });
-    const topDimensions = filteredDimensions.sortBy('score').reverse().slice(0, maxSize);
-    const topDimensionLabels = [...new Set(topDimensions.map(key => key.label.split('=')[1]))];
-
-    // Build the array of subdimension objects for the selected dimension
-    topDimensionLabels.forEach((subDimension) => {
-      if (dimensionObj[subDimension] && subDimension !== '') {
+  // Build the array of subdimension objects for the selected dimension
+  if (dimensionKeys && dimensionKeys.length) {
```

Comment:
unnecessary check here since dimensionKeys should always be truthy since it's the result of `Object.keys(dimensionObj)`. 
```
[] === true;
```

Reply:
Also, do we also need to check for `.length` here since we're calling `.forEach` on an array?

Reply:
Ok, since `Object.keys({}) = []`, I'm ok with removing that.